src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  currentIndex?: number;
+  autoPlay?: boolean;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+const CarouselContext = React.createContext<
+  | {
+      currentIndex: number;
+      setCurrentIndex: (index: number | ((old: number) => void)) => void;
+      slideCount: number;
+      setSlideCount: (length: number | ((old: number) => void)) => void;
+      keysPressed: Record<string, boolean>;
+    }
+  | undefined
+>(undefined);
+
+export const Carousel = (props: CarouselProps) => {
+  const { currentIndex: userCurrentIndex = 0, children, ...rest } = props;
+
+  useTheme();
+
+  const [currentIndex, setCurrentIndex] = React.useState(userCurrentIndex);
+  const [slideCount, setSlideCount] = React.useState(0);
+
+  const [keysPressed, setKeysPressed] = React.useState<Record<string, boolean>>(
+    {},
+  );
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.altKey || event.ctrlKey || event.metaKey) {
+      return;
+    }
+
+    if (event.key === 'ArrowLeft' || event.key === 'ArrowRight') {
+      setKeysPressed((old) => ({ ...old, [event.key]: true }));
```

Comment:
Can you explain this part?

Reply:
I'm saving the keys pressed on key down (and removing on key up) so that I can apply `[data-pressed]` on the left/right navigation buttons. If you try to tab to the carousel and press left/right arrow keys, you will see a visual indicator around the buttons.

Reply:
Is it some standard or what?

Reply:
no, it's the design we decided on

Reply:
I am just wondering whether it is worth having these all handlers.

Reply:
It's needed now because I've moved the state update into keyup to fix a bug when you hold down the key for too long.